packages/-ember-data/tests/integration/relationships/belongs-to-test.js

```diff
@@ -119,6 +119,99 @@ module('integration/relationship/belongs-to BelongsTo Relationships (new-style)'
     assert.ok(personPet === pet, 'We ended up in the same state');
   });
 
+  test('async belongsTo resolves to the record when sideloaded', async function(assert) {
+    let personFindRecordCalls = 0;
+    this.owner.register(
+      'adapter:pet',
+      JSONAPIAdapter.extend({
+        shouldBackgroundReloadRecord() {
+          return false;
+        },
+        findRecord() {
+          assert.ok(false, 'We should not attempt to fetch the pet');
+          // still resolve so that test can continue nicely
+          return resolve({
+            data: {
+              type: 'pet',
+              id: '1',
+              attributes: { name: 'Shen' },
+              relationships: {
+                bestHuman: {
+                  data: { type: 'person', id: '1' },
+                },
+              },
+            },
+          });
+        },
+        findBelongsTo() {
+          assert.ok(false, 'We should not attempt to fetch the person for out pet');
+          return this.store.adapterFor('person').findRecord();
+        },
+      })
+    );
+
+    this.owner.register(
+      'adapter:person',
+      JSONAPIAdapter.extend({
+        shouldBackgroundReloadRecord() {
+          return false;
+        },
+        findRecord() {
+          assert.equal(++personFindRecordCalls, 1, 'We call findRecord only once for our person');
+          return resolve({
+            data: {
+              type: 'person',
+              id: '1',
+              attributes: { name: 'Chris' },
+              relationships: {
+                bestDog: {
+                  data: { type: 'pet', id: '1' },
+                  links: {
+                    related: './pet/1',
+                  },
+                },
+              },
+            },
+            included: [
+              {
+                type: 'pet',
+                id: '1',
+                attributes: { name: 'Shen' },
+                relationships: {
+                  bestHuman: {
+                    data: { type: 'person', id: '1' },
+                  },
+                },
+              },
+            ],
+          });
+        },
+        findBelongsTo() {
+          return this.store.adapterFor('pet').findRecord();
+        },
+      })
+    );
+
+    const person = await store.findRecord('person', '1');
+    const pet = store.peekRecord('pet', '1');
+    const personPetRequest = person.get('bestDog');
+    const personPet = await personPetRequest;
+
+    assert.ok(personPet === pet, 'We ended up in the correct state via .get');
+
+    let promiseDog;
+    await personPetRequest.then(dog => {
+      promiseDog = dog;
+      return dog;
+    });
+
+    assert.ok(promiseDog === pet, 'We ended up in the correct state via .get.then');
+
+    const foundPet = await store.findRecord('pet', '1');
+
+    assert.ok(foundPet === pet, 'We ended up in the correct state via findRecord');
```

Comment:
This is asserting that there wasn't a request made right, there isn't really a failure scenario for peek to return something else than a find, right?

Reply:
no, it is asserting that the result of `findRecord` matches the result of `peekRecord`